src/lib/authenticator.ts

```diff
@@ -54,6 +55,7 @@ export class GenesysCloudClientAuthenticator {
     }
 
     this.config = { ...config } as IAuthenticatorConfig;
+    this.customHeaders = config.customHeaders || {};
```

Comment:
I'm going to fix this in the release PR but this is actually problematic because it will always be truthy.

So we always end up in this conditional:
```
    if (this.customHeaders) {
      Object.entries(this.customHeaders).forEach(([key, value]) => {
        req = req.set(key, value);
      });
    }
```

We should actually only pass in `customHeaders` if we have them from the config and not set a default value. I'm going to remove the `|| {}` and make `customHeaders` an optional field to fix.